src/base/net/downloadhandlerimpl.cpp

```diff
@@ -121,9 +124,13 @@ void DownloadHandlerImpl::processFinishedDownload()
     }
 
     // Success
+#if (QT_VERSION >= QT_VERSION_CHECK(6, 3, 0))
+    m_result.data = m_reply->readAll();
+#else
     m_result.data = (m_reply->rawHeader("Content-Encoding") == "gzip")
                     ? Utils::Gzip::decompress(m_reply->readAll())
```

Comment:
I wonder if we could detect whether the data is really gzipped in `Utils::Gzip::decompress()`.